KMaina/Store-Manager-API
# #161310327 Merge the Products and Users Resources

#### What does this PR do?

Allow a user register, login in the app and view all users

Allow a user to add and view products
#### Description of Task to be completed?

Have the following endpoints working

`POST /api/v1/auth/signup`

`GET /api/v1/users`

`GET /api/v1/users/<int:user_id>`

`POST /api/v1/products`

`GET /api/v1/products`

`GET /api/v1/products/<product_id>`

### How should this be manually tested?

- clone the repo locally

- cd into the Store-Manager-API folder

- run `git pull https://github.com/KMaina/Store-Manager-API.git ch-merge-products-users-161310327`

- in a terminal run the command pytest

- ensure that the following environment variables are set

```
FLASK_ENV=development
FLASK_APP=run.py
SECRET_KEY=<anything-of-choice>
```

- in a terminal run flask run to start the server

- in Postman use the following JSON body to sign up

```
{
   "name" : "Ken",
   "password" : "pass1234",
   "confirm" : "pass1234"
}
```

- in Postman use the following JSON body to log in

```
{
   "name" : "Ken",
   "password" : "pass1234"
}
```

- in Postman use the following JSON body to add a product

```
{
   "name" : "Bread",
   "quantity" : 30,
   "price" : 50,
   "reorder"  : 20 
}
```

#### Any background context you want to provide?

This is all functionality for the users  and products endpoints
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/161310327
